services/server/src/handlers/user.ts

```diff
@@ -189,122 +188,116 @@ export async function create(req: Request, res: Response, next: NextFunction) {
+  if (currentUser.id !== targetUser.id && isRegularUser(currentUser))
+    return next(forbidden());
+
+  // return forbidden if the studio is trying to drop user image
+  if (isStudio(currentUser) && image === null) return next(forbidden());
+
+  // return forbidden if the studio is trying to drop media for an unassociated user
+  if (isStudio(currentUser) && targetTutor?.studioId !== currentUser.id)
+    return next(forbidden());
+
+  if (password) {
+    const expectedPasswordHash = await users.findUserPasswordHash(
+      targetUser.id
+    );
 
-        const isValidCurrentPassword =
-          (password.current === null && expectedPasswordHash === null) ||
-          (password.current &&
-            expectedPasswordHash &&
-            isSamePassword(password.current, expectedPasswordHash));
-        if (!isValidCurrentPassword) return next(wrongPassword());
-
-        const validPassword = isValidPassword(password.new);
-        if (validPassword !== true) return next(apierror(validPassword, 400));
-      }
-
-      // Remove assets from the object store
-      if (image === null && targetUser.image) s3.drop(targetUser.image);
-      if (targetTutor && video === null && targetTutor.video)
-        await s3.drop(targetTutor.video);
-      if (targetTutor && thumbnail === null && targetTutor.thumbnail)
-        await s3.drop(targetTutor.thumbnail);
-
-      // Remove assets ids from the database / update user data
-      const updatedUser = await knex.transaction(
-        async (tx: Knex.Transaction) => {
-          const updatePayload = isStudio(currentUser)
-            ? {}
-            : {
-                city,
-                name,
-                email,
-                gender,
-                birthYear,
-                phone,
-                image,
-                // Reset user verification status incase his email updated.
-                verifiedEmail: email ? false : undefined,
-                password: password ? hashPassword(password.new) : undefined,
-              };
-          const user = await users.update(id, updatePayload, tx);
-
-          const tutorData =
-            bio !== undefined ||
-            about !== undefined ||
-            notice !== undefined ||
-            video === null ||
-            thumbnail === null;
-
-          if (tutorData && targetTutor)
-            await tutors.update(
-              targetUser.id,
-              {
-                bio,
-                about,
-                notice,
-                video,
-                thumbnail,
-              },
-              tx
-            );
-
-          return user;
-        }
+    const isValidCurrentPassword =
+      (password.current === null && expectedPasswordHash === null) ||
+      (password.current &&
+        expectedPasswordHash &&
+        isSamePassword(password.current, expectedPasswordHash));
+    if (!isValidCurrentPassword) return next(wrongPassword());
```

Comment:
Let's simplify the code here.

```suggestion
  const noPassword = password.current === null && expectedPasswordHash === null
  const samePassword = password.current &&
        expectedPasswordHash &&
        isSamePassword(password.current, expectedPasswordHash)
    const isValidCurrentPassword = noPassword || samePassword;
    if (!isValidCurrentPassword) return next(wrongPassword());
```